app/models/mixins/custom_attribute_mixin.rb

```diff
@@ -21,6 +23,28 @@ module CustomAttributeMixin
         miq_custom_set(custom_str, value)
       end
     end
+
+    def self.custom_keys
+      CustomAttribute.where(:resource_type => base_class).map(&:name).uniq
+    end
+
+    def self.load_custom_attributes_for(cols)
+      custom_attributes = CustomAttributeMixin.select_virtual_custom_attributes(cols)
+      custom_attributes.each { |custom_attribute| add_custom_attribute(custom_attribute) }
+    end
+
+    def self.add_custom_attribute(custom_attribute)
+      virtual_column(custom_attribute.to_sym, :type => :string, :uses => :custom_attributes)
+
+      define_method(custom_attribute.to_sym) do
+        custom_attribute_without_prefix = custom_attribute.sub(CUSTOM_ATTRIBUTES_PREFIX, "")
+        custom_attributes.select { |x| custom_attribute_without_prefix.casecmp(x.name.downcase) == 0 }.shift.try(:value)
```

Comment:
> Is the value very complex to determine, or can we do something simple like where("lower(value) = ?", x.name.downcase)?
yes I can use it but I think that it is producing more queries 

I have two VMs and 4 custom attributes what I want to report (names `attr_name_1_1, attr_name_1_3, attr_name_2_1, attr_name_2_3`).

a) with  
```ruby
custom_attributes.where("lower(name) = ?", custom_attribute_without_prefix.downcase).pluck(:value).shift
```
it is generating queries on custom_attributes table
there is log 
http://pastebin.test.redhat.com/389401
so for each VM, 4 queries, so together 8 queries moreover then case b)

b) with 
```ruby
custom_attributes.select { |x| custom_attribute_without_prefix.casecmp(x.name.downcase) == 0 }.shift.try(:value)
```
http://pastebin.test.redhat.com/389404
there are no addition queries to custom_attributes

so what do think which case I should use ?